designs/NuGet-CPPCLI-Support.md

```diff
@@ -0,0 +1,114 @@
+
+When discussing the changes here is a summary from a member on the C++ team.
+
+```md
+TargetFramework*/TargetPlatform* properties for C++/CLI are pretty much the same as for c#, so they should be used in nuget instead of just checking for .vcxproj extension. Also, C++/CLI can use/reference not only managed, but native code as well, so “native” packages should be allowed together with the appropriate managed ones.
+You can use the following properties to distinguish vc projects:
+TargetPlatformIdentifier/Version/Moniker should be defined for all vc projects.
+CLRSupport
+False or empty – native project
+NetCore – C++/CLI .NET Core project
+Everything else (true, safe, etc)  - C++/CLI .NET Framework project
+TargetFramework is only defined for C++/CLI .NET Core projects.
+TargetFrameworkIdentifier/Version/Moniker properties are defined for C++/CLI .NET Framework projects. Note, that they are also confusingly defined for pure native ones too (in msbuild common targets, at least, they used to be), so you need to check other properties first (i.e. TargetPlatformIdentifier and CLRSupport). You can also use project capabilities: “native” vs “managed”
+```
+
+## Who are the customers
+
+Customers using C++/CLI projects.
+
+## Goals
+
+Support the C++ team as necessary to provide NuGet PackageReference support for C++/CLI projects.
+
+At this point focused to:
+
+* Ensure NuGet is interpreting the correct framework.
+
+## Non-Goals
+
+* packages.config support for C++/CLI projects. Focus is on .NET Core scenarios right now.
+
+## Solution Overview
+
+In managed projects, NuGet interprets the effective framework by looking at `TargetFrameworkMoniker`, `TargetPlatformMoniker` and in UWP cases, `TargetPlatformMinVersion`.
+
+C++ projects have an additional property CLRSupport that's set for C++ projects.
+
+Currently the steps for NuGet framework inference for project types still actively supported are as follows:
+
+1. Does the project file extension end with `.vcxproj`? Then `native0.0`
+1. Does `TargetPlatformMoniker` start with `UAP`? Then the framework is `UAP{version}`, where `TargetPlatformMinVersion` is preferred over `TargetPlatformVersion`.
+1. `TargetPlatformMoniker`, unless the framework is `.NETCoreApp` with version greater than 5.0, then it's the combination of `TargetFrameworkMoniker` and `TargetPlatformMoniker`.
+
+To understand the proposed changes, here's a mapping of all `C++` project types and the property value for the appropriate ones.
+
+| Project type | TargetFrameworkIdentifier | TargetPlatformIdentifier | CLRSupport | Effective NuGet framework | Notes | Open questions |
+|--------------|---------------------------|--------------------------|------------|---------------------------|-------|----------------|
+| Native C++ | .NETFramework,Version=v4.0 | Windows,Version=10.0.19041.0 | false | native | NuGet will continue special casing vcxproj. |

```

Comment:
native c++ identifies as .NET Framework 4.0?

Reply:
Native C++ has a default. 
The `CLRSupport` flag will be the first thing read, `false` or `` means native.